File: dcp-solutions/2024-11-13/solution.py
```python
def numKnightTours(N):
    # Initialize all positions as -1 (not visited)
    board = [-1] * (N ** 2)

    # Initialize possible moves
    moves = [(2, 1), (1, 2), (-1, 2), (-2, 1), (-2, -1), (-1, -2), (1, -2), (2, -1)]

    def is_valid(x, y):
        return 0 <= x < N and 0 <= y < N

    def dfs(x, y, step):
        if step == N ** 2:
            return 1
        count = 0
        for dx, dy in moves:
            nx, ny = x + dx, y + dy
            if is_valid(nx, ny) and board[nx * N + ny] == -1:
                board[nx * N + ny] = step
                count += dfs(nx, ny, step + 1)
                board[nx * N + ny] = -1
        return count

    return dfs(0, 0, 1) % 1000000007
```

File: dcp-solutions/2024-11-13/solution.js
```javascript
function numKnight Tours(N) {
    let moves = [[-2, -1], [-2, 1], [-1, -2], [-1, 2], [1, -2], [1, 2], [2, -1], [2, 1]];
    let visited = Array.from({length: N*N}, () => false);
    let result = new Set();

    function dfs(i, j, step) {
        if (step === N*N) {
            result.add(JSON.stringify([i, j]));
            return;
        }

        for (let [dx, dy] of moves) {
            let ni = i + dx;
            let nj = j + dy;

            if (ni >= 0 && ni < N && nj >= 0 && nj < N && !visited[ni*N+nj]) {
                visited[ni*N+nj] = true;
                dfs(ni, nj, step+1);
                visited[ni*N+nj] = false;
            }
        }
    }

    for (let i = 0; i < N; i++) {
        for (let j = 0; j < N; j++) {
            dfs(i, j, 1);
        }
    }

    return result.size;
}
```